google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -192,11 +198,55 @@ def executemany(self, operation, seq_of_parameters):
         for parameters in seq_of_parameters:
             self.execute(operation, parameters)
 
+    def _bqstorage_fetch(self, bqstorage_client, size):
+        """TODO: docstring
+        """
+        # TODO: add assumption that query job exists check has been performed
+
+        # Since a BQ storage client instance is passed in, it means that
+        # bigquery_storage_v1beta1 library is available (no ImportError)
+        from google.cloud import bigquery_storage_v1beta1
+
+        destination = self._query_job.destination
+        table_reference = bigquery.table.TableReference(
+            bigquery.dataset.DatasetReference(
+                destination.project, destination.dataset_id,
+            ),
+            destination.table_id,
+        )
+
+        # TODO: same checks as in _pandas_helpers._download_table_bqstorage()?
+        # (rading from partitions and snapshots)
```

Comment:
DB-API is only used to run queries and download results, so these cases shouldn't happen.

I suppose they could happen if someone writes to a specific partition by manually setting a destination table, but I don't believe we support this in the DB-API.